backend/gsr_booking/views.py

```diff
@@ -275,3 +284,71 @@ def get(self, request):
                 request.user, request.user.booking_groups.filter(name="Penn Labs").first()
             )
         )
+
+
+class GSRShareCodeViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    ViewSet for creating, retrieving, and deleting GSR share codes.
+    """
+
+    serializer_class = GSRShareCodeSerializer
+    lookup_field = "code"
+    lookup_value_regex = r"[A-Za-z0-9_-]{8}"
+
+    permission_classes = [IsShareCodeOwner]
+
+    def get_permissions(self):
+        if self.action == "retrieve":
+            return [AllowAny()]
+        elif self.action == "destroy":
+            return [IsAuthenticated(), IsShareCodeOwner()]
+        return [IsAuthenticated()]
+
+    def get_queryset(self):
+        if self.action == "retrieve":
+            # Allow anyone to retrieve by code
+            return GSRShareCode.objects.all()
+        # For create/destroy, only show user's own share codes
+        return self.request.user.gsr_share_codes.all()
+
+    def get_serializer_class(self):
```

Comment:
nit: if you're using this (which is fine), probably delete the serializer_class var